python/cuml/ensemble/randomforestregressor.pyx

```diff
@@ -0,0 +1,620 @@
+            new RandomForestMetaData[double, double]()
+
+        # rf_forest.trees = state["trees"]
+        if self.dtype == np.float32:
+            print(" dtype is float32 in set state")
+            rf_forest.rf_params = state["rf_params"]
+            state["rf_forest"] = <size_t>rf_forest
+        else:
+            print(" dtype is float64 in set state")
+            rf_forest64.rf_params = state["rf_params64"]
+            state["rf_forest64"] = <size_t>rf_forest64
+
+        self.__dict__.update(state)
+
+    def _get_type(self):
+        if self.criterion == 0:
+            return GINI
+        elif self.criterion == 1:
+            return ENTROPY
+        elif self.criterion == 2:
+            return MSE
+        elif self.criterion == 3:
+            return MAE
+        else:
+            return CRITERION_END
+
+    def _get_max_feat_val(self):
+        if type(self.max_features) == int:
+            max_feature_val = self.max_features/self.n_cols
+        elif type(self.max_features) == float:
+            max_feature_val = self.max_features
+        elif self.max_features == 'sqrt':
+            max_feature_val = 1/np.sqrt(self.n_cols)
+        elif self.max_features == 'log2':
+            max_feature_val = math.log2(self.n_cols)/self.n_cols
+        else:
+            max_feature_val = 1.0
+        return max_feature_val
+
+    def fit(self, X, y):
+        """
+        Perform Random Forest Classification on the input data
+
+        Parameters
+        ----------
+        X : array-like (device or host) shape = (n_samples, n_features)
+            Dense matrix (floats or doubles) of shape (n_samples, n_features).
+            Acceptable formats: cuDF DataFrame, NumPy ndarray, Numba device
+            ndarray, cuda array interface compliant array like CuPy
+        y : array-like (device or host) shape = (n_samples, 1)
+            Dense vector (int32) of shape (n_samples, 1).
+            Acceptable formats: NumPy ndarray, Numba device
+            ndarray, cuda array interface compliant array like CuPy
+            These labels should be contiguous integers from 0 to n_classes.
+        """
+        cdef uintptr_t X_ptr, y_ptr
+        cdef RandomForestMetaData[float, float] *rf_forest = \
+            <RandomForestMetaData[float, float]*><size_t> self.rf_forest
+        cdef RandomForestMetaData[double, double] *rf_forest64 = \
+            <RandomForestMetaData[double, double]*><size_t> self.rf_forest64
+
+        y_m, y_ptr, _, _, y_dtype = input_to_dev_array(y)
+
+        X_m, X_ptr, n_rows, self.n_cols, self.dtype = \
+            input_to_dev_array(X, order='F')
+
+        cdef cumlHandle* handle_ =\
+            <cumlHandle*><size_t>self.handle.getHandle()
+
+        max_feature_val = self._get_max_feat_val()
+        if type(self.min_rows_per_node) == float:
+            self.min_rows_per_node = math.ceil(self.min_rows_per_node*n_rows)
+
+        self.split_criterion = self._get_type()
```

Comment:
Same question as Classifier, isn't this set in the init?

Reply:
the variable `criterion` accepted the entered integer value for the `split_criterion` hyper-parameter but in case the entered value was not present in the list of accepted options we would assign it the option `CRITERION_END`